### ISSUE TITLE
Prettier 1.18 breaks support for TS users < 3.3

### ISSUE THREAD

**Issue by mattlewis92** [prettier/prettier#6197]
**Prettier 1.18.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEqEE+AFAIYBOA5gM5L4DaAOlPux+7TJTlNQBpWnTt178hbEVx59Bw6fjFzJipbIkLpyzVO0b5ekTsNqTqxea3GD19gF0AlPmABfEAJAQADrmi1kUCpKCAB3AAUqBACUcgA3CBwAEw8QACNKcjAAazgYAGVvLLlkHnQ4TwALGABbABsAdUqceFoisDh86JacOJaAT2RwWgDPPlo4ShhwzOoa8mRUcjqJzwArWgAPACFMnLz88hq4ABk+OEXl1ZANzfy5OrgARXQIeEuVipAiygnKIZg-W8cFoYF4vlS3nEMAayRglWQAA4AAyeKEQCYNTLeIZQkGTOIXTwAR1e8BmPhiIHItAAtFA4HAkkzUpQ4KScGyZuQ5gskEtPp4JjUcKVKOUhQ9nmSLvyrl8YOQ0rCkvDkAAmTw8cg4OpyADCEBq8yGIIArKl0BMACpKmICiauVxAA)
```sh
--parser typescript
--trailing-comma es5
```

**Input:**
```tsx
function foo (args: [
        string,
        string,
        string,
        string,
        string,
        string,
        string,
        string,
        string,
        string,
        string
      ]) {}
```

**Output:**
```tsx
function foo(
  args: [
    string,
    string,
    string,
    string,
    string,
    string,
    string,
    string,
    string,
    string,
    string,
  ]
) {}

```

**Expected behavior:**

According to the PR at https://github.com/prettier/prettier/pull/6172, this should only be applied when the trailing comma is `all` but it also gets applied when it's set to `es5`.

Secondly because we were on TS 3.2 (we can't upgrade get because we're on an older angular version) this broke our build as TS 3.2 can't parse the output. I think the TS version could be read by prettier and then used to determine whether to apply this transformation.

**Comment by duailibe** [comment]
@mattlewis92 

Oh thanks for the report! Indeed there was a bug there. I'll fix it and release 1.18.2 in a bit.

> I think the TS version could be read by prettier and then used to determine whether to apply this transformation.

This is a good idea.

**Comment by mattlewis92** [comment]
Awesome, thanks!
